### PR TITLE
Fix hash of NCube & add plotting utilities

### PR DESCRIPTION
**Pull request by anlavandier**
* Bugfix: change the `__hash__` attribute of `NCubeInterior`
* Add a return value of `None` when `Domain.get_subdomain` is called with the empty tuple
* Add plotting functions in `utilities.utils.py`
* Update library version to 0.15.2

### CODE REVIEW

**Comment by yguclu** [comment]
Thanks for this PR! Can you please provide a short summary of the changes made?

**Comment by anlavandier** [comment]
I updated the main comment.

File: sympde/utilities/utils.py [hunk 1]
```diff
@@ -79,3 +84,92 @@ def f_vec_v(*XYZ):
...
+        *[np.linspace(patch.min_coords[i],
+                      patch.max_coords[i],
+                      num=refinement,
+                      endpoint=True) for i in range(2)],
+        indexing='ij',
+    )
+    XX, YY = map_call(*mesh_grid)
+
+    ax.plot(XX[:, ::5], YY[:, ::5], 'k')
+    ax.plot(XX[::5, :].T, YY[::5, :].T, 'k')
```
Comment (yguclu) [inline sympde/utilities/utils.py:175]: No newline at end of file

**Review by yguclu — CHANGES_REQUESTED** [review]
Good job! I have a couple of request regarding file `sympde.utilities/utils.py`:

- Please add some docstrings to the new functions 
- Given that it is not convenient to add unit tests for visualization functions, please add a short example of usage, which can be included at the bottom of the file under the section `if __name__ == '__main__':`

**Review by yguclu — CHANGES_REQUESTED** [review]
Please add labels to the `x`, `y`, and `z` axes.

File: sympde/utilities/utils.py [hunk 2]
```diff
@@ -79,3 +85,206 @@ def f_vec_v(*XYZ):
...
+    Parameters
+    ----------
+    patch : sympde.topology.InteriorDomain
+
+    mapping : sympde.topology.mapping
+
+    ax : mpl_toolkits.mplot3d.axes3d.Axes3D
+        Axes object on which the patch is drawn.
+
+    refinement : int, default=15
```
Comment (yguclu) [inline sympde/utilities/utils.py:198]: Please add "Number of straight line segments used to approximate each boundary edge."

File: sympde/utilities/utils.py [hunk 3]
```diff
@@ -79,3 +85,206 @@ def f_vec_v(*XYZ):
...
+
+    mapping : sympde.topology.mapping
+
+    ax : matplotlib.axes.Axes
+        Axes object on which the patch is drawn.
+
+    isolines : bool, default=False
+        If true also plots some iso-lines
+
+    refinement : int, default=40
```
Comment (yguclu) [inline sympde/utilities/utils.py:256]: See above

**Review by yguclu — CHANGES_REQUESTED** [review]

File: sympde/utilities/utils.py (hunk 2 above)
Comment (anlavandier) [reply sympde/utilities/utils.py:205]: Done in [825c1fa](https://github.com/pyccel/sympde/pull/112/commits/825c1fa5adee918fbc3b8f56eb501853705865a2)

File: sympde/utilities/utils.py (hunk 3 above)
Comment (anlavandier) [reply sympde/utilities/utils.py:264]: Done in [825c1fa](https://github.com/pyccel/sympde/pull/112/commits/825c1fa5adee918fbc3b8f56eb501853705865a2)

**Review by yguclu — APPROVED** [review]

**Pull request merged**